core/src/main/scala/org/apache/pekko/persistence/jdbc/state/scaladsl/JdbcDurableStateStore.scala

```diff
@@ -113,21 +112,34 @@ class JdbcDurableStateStore[A](
       }
   }
 
-  def deleteObject(persistenceId: String): Future[Done] =
+  override def deleteObject(persistenceId: String): Future[Done] =
     db.run(queries.deleteFromDb(persistenceId).map(_ => Done))
 
-  def deleteObject(persistenceId: String, revision: Long): Future[Done] =
-    db.run(queries.deleteFromDb(persistenceId).map(_ => Done))
+  override def deleteObject(persistenceId: String, revision: Long): Future[Done] = {
+    db.run(queries.deleteBasedOnPersistenceIdAndRevision(persistenceId, revision)).map { count =>
+      {
+        if (count == 0) {
+          // if you run this code with Pekko 1.0.x, no exception will be thrown here
+          // this matches the behavior of pekko-connectors-jdbc 1.0.x
+          // if you run this code with Pekko 1.1.x, a DeleteRevisionException will be thrown here
+          DurableStateExceptionSupport.createDeleteRevisionExceptionIfSupported(
+            s"Failed to delete object with persistenceId [$persistenceId] and revision [$revision]")
+            .foreach(throw _)
+        }
+        Done
+      }
+    }
```

Comment:
Set the execution context to be samethread

Reply:
changed to use parasitic context